classes/02class/exercises/c02-network07/gpavelar/README.md

```diff
@@ -0,0 +1,170 @@
+# Output
+An error occurred (InvalidGatewayID.NotFound) when calling the CreateRoute operation: The gateway ID 'nat-0dcad8a11bc5686fa' does not exist
+## I've already delted the natgatway created previously. But, Let's create it again
+
+### Creating NAT AGAIN
+## Allocate IP
+aws ec2 allocate-address --domain vpc
+
+# Output
+eipalloc-0711caa72a2042385      vpc     ap-southeast-2  52.62.142.50    amazon
+
+## SubnetId = subnet-08da4676baf658cb2
+## allocation-id = eipalloc-0711caa72a2042385
+## Create NAT gateway
+aws ec2 create-nat-gateway --allocation-id eipalloc-0711caa72a2042385  --subnet-id subnet-08da4676baf658cb2 --tag-specifications 'ResourceType="natgateway",Tags=[{Key="Name",Value="devopsacademy-ngw"}]'
+
+# Output
+7d2dc0a8-1b25-45ab-a711-b36b08e5f353
+NATGATEWAY      2020-07-06T15:54:11+00:00       nat-0c2e8f4ea792c8a75   pending subnet-08da4676baf658cb2        vpc-074c260483405f2bb
+NATGATEWAYADDRESSES     eipalloc-0711caa72a2042385
+TAGS    Name    devopsacademy-ngw
+
+aws ec2 delete-nat-gateway --nat-gateway-id nat-0c2e8f4ea792c8a75
+# Output
+nat-0c2e8f4ea792c8a75
+
+## Perform the command again
+aws ec2 create-route --route-table-id rtb-040e043e6f45c836c --destination-cidr-block 0.0.0.0/0 --gateway-id nat-0c2e8f4ea792c8a75
+# Output
+True
+```
+
+- Answer the following questions:
+
+  - How did you accomplish both instructions?
+
+    ```
+    I think so.
+    ```
+
+  - Why did you add 0.0.0.0 route to the IGW on public route table?
+
+    ```
+    Through this steps I enable my subnets to be public.
+    ```
+
+  - Why did you add 0.0.0.0 route to the NGW on private route table?
+
+    ```
+    Allow a way to connect my subnets securely over ther internet. Their IP addresses should definetely not go public. So we use NAT.
+    ```
+
+  - What is the difference of IGW and NGW?
+
+    ````
+    Basically, what I have got it, Internet Gateway is a way that my subnet has to communicate beyond the vpc (acessing internet). On the other hand, NAT just is a bridge betwwen your private subnet and the internet or any aws services.  ```
+
+    ````
+
+  - Can you delete the destination route to your VPC network? Why?
+
+    ```
+    You are not able to delete main route table.
+    If you create a custom subnet route table, You are able to delete just when there is no association in a custom route table.
+    ```
+
+  - What happens if you do not associate your route table with any subnets?
+
+    ```
+    Usually, It would not be able to communicate with others in the subnet. But AWS has a default/main table route that allow you use
+    ```
+
+* Any extra challenges faced?
+
+> Link one by one. =D
+
+> When I was performing `aws ec2 release-address --allocation-id eipalloc-0711caa72a2042385`. I got the following error
+
+```bash
+  An error occurred (AuthFailure) when calling the ReleaseAddress operation: You do not have permission to access the specified resource.
+```
+
+Ufa! After 200x attempts, i got it !.
```

Comment:
Maybe it was still linked with something. You can also use the console if you're in doubt:

https://ap-southeast-2.console.aws.amazon.com/ec2/v2/home?region=ap-southeast-2#Addresses:

Reply:
Yeah, I've removed any link before. I think it was during the night (2a.m). Could it be any maintenance?